ee/vellum_ee/workflows/display/nodes/vellum/conditional_node.py

```diff
@@ -40,17 +40,17 @@
 
 @dataclass
 class RuleIdMap:
-    id: UUID
+    id: Union[UUID, str]
     lhs: Optional["RuleIdMap"]
     rhs: Optional["RuleIdMap"]
-    field_node_input_id: Optional[UUID]
-    value_node_input_id: Optional[UUID]
+    field_node_input_id: Union[Optional[UUID], Optional[str]]
+    value_node_input_id: Union[Optional[UUID], Optional[str]]
```

Comment:
nit: I could see us using only strings here in `RuleIdMap` instead of a union between uuid and string. The main thing I wanted to avoid was having to change `node_input_ids_by_name: ClassVar[Dict[str, UUID]] = {}` to `node_input_ids_by_name: ClassVar[Dict[str, str]] = {}`, which you've achieved above with `BaseNodeVellumDisplay`. If you only ever use strings in `RuleIdMap`, it may simplify some codegen code too.